artiq/protocols/sync_struct.py

```diff
@@ -154,7 +156,12 @@ def __init__(self, notifier_name, target_builder, notify_cb=None,
                     process_mod(target, mod)
 
                 for notify_cb in self.notify_cbs:
-                    notify_cb(mod)
+                    try:
+                        notify_cb(mod)
+                    except Exception as e:
+                        logger.error("Exception in notifier callback: {}"
+                                     .format(e))
+                        raise ConnectionError
```

Comment:
raise ConnectionError?

Reply:
Maybe that's a bit of a hack. The intention was to ensure that the Subscriber gets closed when there is an exception in a callback.

Another way of achieving the same is to replace `except ConnectionError` with `Except Exception` in the original except statement.

Reply:
It's a hack. Please find a clean solution.

Reply:
happy with just expanding the original `except` block to `except Exception`? Or, even `except:` to cover cases like `KeyboardInterrupt` during a notifier cb?

Reply:
On second thoughts, I'd prefer to keep a separate exception block for the callback so that one gets a message in the log when a callback raises an exception, rather than silently dropping the connection.

Refactored to remove the hack.